react/features/base/tracks/middleware.js

```diff
@@ -37,19 +42,55 @@ MiddlewareRegistry.register(store => next => action => {
         _setMuted(store, action, MEDIA_TYPE.AUDIO);
         break;
 
-    case SET_CAMERA_FACING_MODE:
+    case SET_CAMERA_FACING_MODE: {
+        // XXX Destroy the local video track before creating a new one or
+        // react-native-webrtc may be slow or get stuck when opening a (video)
+        // capturer twice.
+        const localTrack = _getLocalTrack(store, MEDIA_TYPE.VIDEO);
+
+        if (localTrack) {
+            store.dispatch(_disposeAndRemoveTracks([ localTrack.jitsiTrack ]));
+        }
+
         store.dispatch(
             createLocalTracks({
                 devices: [ MEDIA_TYPE.VIDEO ],
                 facingMode: action.cameraFacingMode
             })
         );
         break;
+    }
 
     case SET_VIDEO_MUTED:
         _setMuted(store, action, MEDIA_TYPE.VIDEO);
         break;
 
+    case TOGGLE_CAMERA_FACING_MODE: {
+        const localTrack = _getLocalTrack(store, MEDIA_TYPE.VIDEO);
+
+        if (localTrack
+                && localTrack.jitsiTrack
+                && localTrack.jitsiTrack.track) {
+            // XXX MediaStreamTrack._switchCamera a custom function implemented
+            // in react-native-webrtc for video which switches between the
+            // cameras via a native WebRTC library implementation without making
+            // any changes to the track.
+            localTrack.jitsiTrack.track._switchCamera();
+
+            // Don't show mirrored video for the back camera.
+            // XXX Relies on the fact that we always open the camera in
+            // user-facing mode first.
+            store.dispatch({
+                type: TRACK_UPDATED,
+                track: {
+                    jitsiTrack: localTrack.jitsiTrack,
+                    mirrorVideo: !localTrack.mirrorVideo
+                }
+            });
```

Comment:
I considered that, but I discarded it because it felt too mobile specific. I feel the same way about `getCameraFacingMode`, come to think of it.